packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts

```diff
@@ -6,13 +6,13 @@ import type { ParseSettings } from './index';
  * This needs to be kept in sync with /docs/users/Versioning.mdx
  * in the typescript-eslint monorepo
  */
-const SUPPORTED_TYPESCRIPT_VERSIONS = '>=4.3.5 <5.3.0';
+const SUPPORTED_TYPESCRIPT_VERSIONS = '>=4.3.5 <5.4.0';
```

Comment:
look at this guy bumping early
we probably want to leave this at 5.3 until the final release and we've generated our types.